upload/modules/Store/gateways/PayPalBusiness/gateway.php

```diff
@@ -29,34 +29,71 @@ public function processOrder(Order $order): void {
+            $transaction = new \PayPal\Api\Transaction();
+            $transaction->setAmount($amount);
+            $transaction->setDescription($order->getDescription());
+            $transaction->setInvoiceNumber($order->data()->id);
+            $transaction->setPurchaseOrder($order->data()->id);
+            $transaction->setCustom($order->data()->id);
 
-        $redirectUrls = new \PayPal\Api\RedirectUrls();
-        $redirectUrls->setReturnUrl(rtrim(URL::getSelfURL(), '/') . URL::build('/store/process/', 'gateway=PayPalBusiness&do=success'))
-            ->setCancelUrl(rtrim(URL::getSelfURL(), '/') . URL::build('/store/process/', 'gateway=PayPalBusiness&do=cancel'));
+            $redirectUrls = new \PayPal\Api\RedirectUrls();
+            $redirectUrls->setReturnUrl(rtrim(URL::getSelfURL(), '/') . URL::build('/store/process/', 'gateway=PayPalBusiness&do=success'))
+                ->setCancelUrl(rtrim(URL::getSelfURL(), '/') . URL::build('/store/process/', 'gateway=PayPalBusiness&do=cancel'));
 
-        $payment = new \PayPal\Api\Payment();
-        $payment->setIntent('sale')
-            ->setPayer($payer)
-            ->setTransactions([$transaction])
-            ->setRedirectUrls($redirectUrls);
+            $payment = new \PayPal\Api\Payment();
+            $payment->setIntent('sale')
+                ->setPayer($payer)
+                ->setTransactions([$transaction])
+                ->setRedirectUrls($redirectUrls);
 
-        try {
-            $payment->create($apiContext);
+            try {
+                $payment->create($apiContext);
+
+                Redirect::to($payment->getApprovalLink());
+            } catch (\PayPal\Exception\PayPalConnectionException $ex) {
+                ErrorHandler::logCustomError($ex->getData());
+            }
+        } else {
+            // Payment subscription
+            $product = null;
+            foreach ($order->getProducts() as $item) {
+                $product = $item;
+                break;
+            }
+
+            $agreement = new \PayPal\Api\Agreement();
+            $agreement->setName($product->data()->name)
+                ->setDescription($product->data()->name)
+                ->setStartDate(date("c", strtotime("+5 mins")));
+
+
+            $agreement->setPlan($this->getPlan($product));
+
+            $payer = new \PayPal\Api\Payer();
+            $payer->setPaymentMethod('paypal');
+            $agreement->setPayer($payer);
+
+            try {
+                $agreement = $agreement->create($apiContext);
 
-            Redirect::to($payment->getApprovalLink());
-        } catch (\PayPal\Exception\PayPalConnectionException $ex) {
-            ErrorHandler::logCustomError($ex->getData());
+                $approvalUrl = $agreement->getApprovalLink();
+
+            } catch (Exception $ex) {
+                echo "Failed to get activate";
+                exit();
+            }
+
+            header("Location:" . $approvalUrl);
+            exit();
```

Comment:
Use `return;` or just nothing at all, seeing as it's at the end anyways this seems redundant?